File: src/main/java/dev/retrotv/crypt/owe/hash/sha/SHA512256.java
package dev.retrotv.crypt.owe.hash.sha;

import dev.retrotv.crypt.owe.hash.Hash;
import dev.retrotv.utils.MessageDigestEncodeUtil;
import dev.retrotv.utils.EncodeUtil;
import lombok.NonNull;

import static dev.retrotv.enums.HashAlgorithm.SHA512256;

/**
 * SHA-512/256 알고리즘으로 암호화 하기 위한 {@link MessageDigestEncodeUtil} 추상 클래스의 구현체 입니다.
 *
 * @author  yjj8353
 * @since   1.8
 */
public class SHA512256 extends Hash {

    @Override
    public String hash(@NonNull byte[] data) {
        return EncodeUtil.binaryToHex(MessageDigestEncodeUtil.encode(SHA512256, data));
    }
}

File: src/main/kotlin/dev/retrotv/crypto/owe/hash/sha/SHA512256.kt
package dev.retrotv.crypto.owe.hash.sha

import dev.retrotv.crypto.owe.hash.HashAlgorithm
import dev.retrotv.data.enums.EncodeFormat
import dev.retrotv.data.utils.toHexString
import dev.retrotv.enums.Algorithm.Hash.SHA512256
import dev.retrotv.utils.digest
import dev.retrotv.utils.encode

/**
 * SHA-512/256 알고리즘으로 암호화 하기 위한 [HashAlgorithm] 추상 클래스의 구현체 입니다.
 *
 * @author  yjj8353
 * @since   1.0.0
 */
class SHA512256 : HashAlgorithm() {
    override fun hash(data: ByteArray): String {
        return toHexString(digest(SHA512256, data))
    }

    override fun hash(data: ByteArray, encodeFormat: EncodeFormat): String {
        return encode(encodeFormat, digest(SHA512256, data))
    }
}
